src/linechart/LineChart.jsx

```diff
@@ -19,35 +14,31 @@ module.exports = React.createClass({
   displayName: 'LineChart',
 
   propTypes: {
-    margins:           React.PropTypes.object,
-    circleRadius:       React.PropTypes.number,
-    displayDataPoints: React.PropTypes.bool,
-    hoverAnimation:    React.PropTypes.bool,
-    interpolate:       React.PropTypes.bool,
-    interpolationType: React.PropTypes.string
-  },
+    circleRadius:   React.PropTypes.number,
+    hoverAnimation: React.PropTypes.bool,
+    margins:        React.PropTypes.object,
+ },
 
   getDefaultProps() {
     return {
-      margins:           {top: 10, right: 20, bottom: 40, left: 45},
-      className:         'rd3-linechart',
-      circleRadius:       3,
-      interpolate:       false,
-      interpolationType: null,
-      displayDataPoints: true,
-      hoverAnimation:    true
+      circleRadius:    3,
+      className: 'rd3-linechart',
+      hoverAnimation: true,
+      margins:        {top: 10, right: 20, bottom: 50, left: 45},
+      xAxisClassName: 'rd3-linechart-xaxis',
+      yAxisClassName: 'rd3-linechart-yaxis',
     };
   },
 
-  render() {
+  _calculateScales: utils.calculateScales,
 
-    var structure = immstruct('lineChart', { voronoi: {}, voronoiSeries: {}});
+  render() {
 
     var props = this.props;
 
-    var data = props.data;
-
-    var interpolationType = props.interpolationType || (props.interpolate ? 'cardinal' : 'linear');
+    if (this.props.data && this.props.data.length < 1) {
```

Comment:
You already dereferenced `props` above, so just make this:

`if (props.data && props.data.length < 1)`